Tasks/ServiceFabricUpdateManifests/Get-VstsBuild.psm1

```diff
@@ -63,28 +67,40 @@
                 $agentTmpFolder = Join-Path (Get-VstsTaskVariable -Name Agent.WorkFolder -Require) "tmp"
                 $artifactZipFile = Join-Path $agentTmpFolder "$PkgArtifactName.zip"
                 $artifactPath = Join-Path $agentTmpFolder $PkgArtifactName
-
+				$downloadArtifact = $true
+				
                 if (Test-Path $artifactPath)
                 {
-                    # If a previous artifact with the same name was already downloaded to the agent's temp folder, delete it
-                    Remove-Item $artifactPath -Recurse -Force | Out-Null
+					if ($OverwritePkgArtifact -eq $true)
+					{
+						# If a previous artifact with the same name was already downloaded to the agent's temp folder, delete it
+						Remove-Item $artifactPath -Recurse -Force | Out-Null
+					}
+					else
+					{
+						# Use prior download of artifact
+						$downloadArtifact = $false
```

Comment:
Have you verified that this artifact would never be a different one than expected. For instance if using a custom agent and you ran the build twice would the second run end up using the same artifact as the first?

Reply:
FYI based on my previous inspection of an agent during runs, the artifacts are not versioned in the temp folder, so my concern is the artifact form build 1 and build 2 having the same name and the artifact form build 2 not getting downloaded b/c the artifact from build 1 is present.

Reply:
The scenario that I have is one where multiple applications are being versioned and I'm not using parallel builds. In my build definition, I would have the first task overwrite the package and the others use it if it exists. In the scenario of the default Service Fabric template having 1 application, it should always be set to overwrite.

Reply:
This would work for your particular case, but would be more useful to the general population with versioned artifacts. It would enable:

- Not needing to ensure the first Update task overwrites the artifact in cases like yours
- Reusing the artifact across builds when building against a fixed build number
- Reusing the artifact across builds when the build fails

Reply:
@NCarlsonMSFT How is the temp directory cleaned up if the artifact is left behind for the next build to use? Could that cause the disk to run out of space? How would you know that the artifact was fully downloaded in the prior build?